addons/survey/tests/test_survey.py

```diff
@@ -296,3 +296,52 @@ def get_question_by_title(survey, title):
         self.assertNotEqual(q_food_not_vegetarian_cloned.triggering_question_id.id, q_is_vegetarian.id)
         self.assertNotEqual(q_food_not_vegetarian_cloned.triggering_answer_id.id,
                             q_is_vegetarian.suggested_answer_ids[1].id)
+
+    def test_get_pages_and_questions_to_show(self):
+        """ Tests the method `_get_pages_and_questions_to_show`. This method takes a
+            recordset of question.question from a survey.survey and returns a recordset without
+            invalid conditional questions and pages without description """
+
+        survey_feedback = self.env.ref('survey.survey_feedback')
+        invalid_records = self.env['survey.question']
+        page_without_description = self.env['survey.question'].create({
+            'title': 'no desc',
+            'survey_id': survey_feedback.id,
+            'sequence': 99,
+            'question_type': False,
+            'is_page': True,
+            'description': False,
+        })
+        question_and_page_ids = survey_feedback.question_and_page_ids
+        returned_questions_and_pages = survey_feedback._get_pages_and_questions_to_show()
+
+        # Only the empty page is an invalid record
+        invalid_records |= page_without_description
+        self.assertEqual(question_and_page_ids - invalid_records, returned_questions_and_pages)
+
+        # Valid conditional question
+        p2_q1 = self.env.ref('survey.survey_feedback_p2_q1')
+        p2_q1.write({
+            'is_conditional': True,
+            'triggering_question_id': self.env.ref('survey.survey_feedback_p1_q3'),
+            'triggering_answer_id': self.env.ref('survey.survey_feedback_p1_q3_sug1'),
+        })
+        returned_questions_and_pages = survey_feedback._get_pages_and_questions_to_show()
+        self.assertEqual(question_and_page_ids - invalid_records, returned_questions_and_pages)
+
+        # Invalid conditional question
+        p2_q1.sequence = 0
+        invalid_records |= p2_q1
+        returned_questions_and_pages = survey_feedback._get_pages_and_questions_to_show()
+        self.assertEqual(question_and_page_ids - invalid_records, returned_questions_and_pages)
+
+        # Question is placed after trigger but trigger is an invalid question
+        p1_q3 = self.env.ref('survey.survey_feedback_p1_q3')
+        p1_q3.write({
+            'is_conditional': True,
+            'triggering_question_id': p2_q1,
+            'triggering_answer_id': self.env.ref('survey.survey_feedback_p2_q1_sug1'),
+        })
+        invalid_records |= p1_q3
+        returned_questions_and_pages = survey_feedback._get_pages_and_questions_to_show()
+        self.assertEqual(question_and_page_ids - invalid_records, returned_questions_and_pages)
```

Comment:
You seem to only check sequence stuff, could add tests cases for:
- question that is not simple / multi choice
- question that has no suggested answers